pulser-core/pulser/register/base_register.py

```diff
@@ -258,7 +265,7 @@ def _to_dict(self) -> dict[str, Any]:
         return obj_to_dict(
             self,
             cls_dict,
-            [np.ndarray.tolist(qubit_coords) for qubit_coords in self._coords],
+            [qubit_coords.as_list() for qubit_coords in self._coords],
```

Comment:
Does this fail for tensors that require grad?

Reply:
I changed this line not due to functionality issues but `mypy` error.

Reply:
I get that, but I was just wondering if `tolist()` failed for tensors that required grad. I just checked that it just doesn't care